fineract-provider/src/main/java/org/apache/fineract/portfolio/savings/domain/SavingsAccountTransaction.java

```diff
@@ -365,6 +369,7 @@ private SavingsAccountTransaction(final Integer transactionTypeEnum, final Local
         this.balanceNumberOfDays = balanceNumberOfDays;
         this.overdraftAmount = overdraftAmount;
         this.createdDate = createdDate;
+        this.submittedOnDate = DateUtils.getBusinessLocalDate();
```

Comment:
Please check the copy transaction logic (`copyTransaction` method) as well, whether it gets properly submitted on date or not!